### PR TITLE
Push docker images with tag prefix

### LINKED ISSUE #1610: Create docker images of version releases
For easy access to debugging and end-users

### PR DESCRIPTION
**Pull request by mscroggs**
When a tag is created, this will push docker images with that tag (eg `dolfinx/dolfinx:v0.1.1-arm64` `dolfinx/lab:v0.1.1-amd64` etc).

Fixes #1610

### CODE REVIEW

File: .github/workflows/docker-end-user.yml [hunk 1]
```diff
@@ -8,6 +8,10 @@ on:
   #push:
   #  branches:
   #    - "**"
+  push:
+    # This will run every time a version tag is created
+    tags:
+      - 'v*'
```
Comment (jhale) [inline .github/workflows/docker-end-user.yml:14]: Are we going to use the `v` prefix? It does seem to be the custom on GitHub. @chrisrichardson?

**Review by jhale — COMMENTED** [review]
Thanks this is really helpful. It's also useful to have the base development images around. Could you add the same trick there?

https://github.com/FEniCS/dolfinx/blob/main/.github/workflows/docker-build-env.yml

File: .github/workflows/docker-end-user.yml (hunk 1 above)
Comment (mscroggs) [reply .github/workflows/docker-end-user.yml:14]: Good spot. The first release doesn't have a v. I guess we can actually just make this push for any tags

Comment (mscroggs) [reply .github/workflows/docker-end-user.yml]: I've remove the v prefix

Comment (jhale) [reply .github/workflows/docker-end-user.yml]: Well, I guess that's the reason to have `v`, to distinguish a release from any other tag. I think @chrisrichardson decided not to have `v` for a reason but I forgot why. 

On balance, I think we can trust people not to push tags.

**Pull request merged**